syntax = "proto3";

option java_package = "ex.grpc";

package wifs;

service WIFS {
  rpc wifs_READ(ReadReq) returns (ReadRes){}
  rpc wifs_WRITE(WriteReq) returns (WriteRes){}
  rpc wifs_INIT(ClientInitReq) returns (ClientInitRes){}
}

message ClientInitReq {

}

message ClientInitRes{
  string primary_ip = 1;
}

message WriteReq {
  int32 address = 1;
  bytes buf = 2;
}

message WriteRes {
  int32 status = 1;
  bytes buf = 2;
}

message ReadReq {
  int32 address = 1;
  bytes buf = 2;
}

message ReadRes {
  int32 status = 1;
  bytes buf = 2;
}
